src/core/modules/impl/ContractDefinitionLoader.ts

```diff
@@ -50,6 +51,10 @@ export class ContractDefinitionLoader implements DefinitionLoader {
     this.logger.debug('Tags decoding', benchmark.elapsed());
     benchmark.reset();
 
+    const contractSrcTx = await this.arweaveWrapper.tx(contractSrcTxId);
+    const contractType: ContractType =
+      getTag(contractSrcTx, SmartWeaveTags.CONTENT_TYPE) == 'application/javascript' ? 'js' : 'wasm';
```

Comment:
Shouldn't we be more careful checking if CONTENT_TYPE == 'application/wasm');

Reply:
👍